File: convert-date/ConvertDate.c
#include <stdio.h>
#include <string.h>
#define MAX 40

int main() {
  char usDate[MAX];
  fgets(usDate, MAX, stdin);
  
  int month = 0, day, year;
  
  if (strlen(usDate) > 10) {
    char monthName[12];
    sscanf(usDate, "%s %d, %d", monthName, &day, &year);
    char *months[] = {"January", "February", "March", "April", "May", "June", 
  	     "July", "August", "September", "October", "November", "December"};
    while (strcmp(monthName, months[month++]) && month < 12);
  }
  else sscanf(usDate, "%d/%d/%d", &month, &day, &year);
  
  printf("%d/%d/%d", day, month, year);
  return 0;
}

File: convert-date/ConvertDate.py
import re
months = {'January': 1, 'February': 2, 'March': 3, 'April': 4, 'May': 5, 'June': 6,
          'July': 7, 'August': 8, 'September': 9, 'October': 10, 'November': 11, 'December': 12}

usDate = re.search(r'(.*)[ /](\d{1,2})[,/] ?(\d{4})', input())

month = usDate.group(1)
day = usDate.group(2)
year = usDate.group(3)

if len(month) > 2:
    month = months.get(month)

print('{}/{}/{}'.format(day, month, year))
